### DISCUSSION TITLE
Jrenc的React+Next.js学习笔记

### DISCUSSION THREAD

**Discussion by jrenc2002** [openbuildxyz/Web3-Frontend-Bootcamp#239] [Materials]
飞书链接：https://sbrs8lf797.feishu.cn/wiki/GYX6wciIwiIWCWk5qgQc95tIntf   
密码：21#2A418

**Comment by linghuccc** [comment]
能解释一下，为什么“（狗都不用）”么？

看了别人的回答，我还以为学到了新东西 (虽然回头看了一下我自己的 code，发现类似的场景，我自己并没有用 useContext )

**Reply by jrenc2002** [reply]
口嗨，最开始自己做着笔记玩的。可以看看GPT的回答。
![image](https://github.com/openbuildxyz/Web3-Frontend-Bootcamp/assets/87536202/1b293e57-1488-465d-86a0-dca66db506bc)
useContext 是 React 中用于共享全局状态的一个钩子，但在一些场景下，它可能并不是最佳选择。以下是一些常见的问题和局限性：

性能问题：

useContext 会在上下文值变化时使所有使用该上下文的组件重新渲染。对于大型应用或频繁更新的状态，这可能会导致性能问题，因为即使只有部分组件需要更新，所有依赖该上下文的组件都会重新渲染。
难以追踪的状态变更：

当使用多个上下文时，代码可能变得复杂且难以维护。多个上下文值的变更和不同组件的重渲染顺序可能难以追踪和调试。
无法选择性订阅：

useContext 没有提供选择性订阅功能。你无法只订阅上下文值的一部分变化。任何值的变化都会触发所有订阅该上下文的组件重新渲染。
测试复杂性：

在测试组件时，需要对每个组件的上下文进行模拟。对于依赖多个上下文的组件，测试代码会变得更加复杂。
替代方案
Redux：

Redux 是一个流行的状态管理库，提供了更加细粒度的状态管理和优化选项。它通过 action 和 reducer 来处理状态变更，并且具有中间件支持，适用于复杂的应用。
Recoil：

Recoil 是 Facebook 开源的一个状态管理库，提供了类似于 React Hooks 的 API，并且支持选择性订阅和高效的状态更新。
MobX：

MobX 是另一个强大的状态管理库，支持响应式编程，自动跟踪状态依赖，优化性能。
Zustand：

Zustand 是一个轻量级但功能强大的状态管理库，提供了简单的 API 和灵活的状态管理，适用于中小型应用。
结论
虽然 useContext 在某些情况下可以简化状态管理，但对于复杂应用或者频繁更新的状态，它可能并不是最佳选择。根据具体需求，选择合适的状态管理工具可以更好地提升应用性能和代码可维护性。
